lib/astutils.cpp

```diff
@@ -1240,7 +1240,7 @@ struct FwdAnalysis::Result FwdAnalysis::checkRecursive(const Token *expr, const
             return Result(Result::Type::BAILOUT);
         }
 
-        if (expr->isName() && Token::Match(tok, "%name% (") && tok->str().find("<") != std::string::npos && tok->str().find(expr->str()) != std::string::npos)
+        if (expr->isName() && Token::Match(tok, "%name% (") && tok->str().find('<') != std::string::npos && tok->str().find(expr->str()) != std::string::npos)
```

Comment:
I have read somewhere that this would make the code faster and that sounds good in theory. However a few months ago I benchmarked on a small program when a char literal is added to a stream and did not see a speed improvement. Have you done benchmarks?

For readability I am guessing that it would be better with string literals here and there in our code.

Reply:
In this case it's a `find()` - I case simply comparing characters instead of doing a `compare()` call might be faster deepening on the length of the string. I did not do any benchmarks. The Lr will probably drop, but I doubt there will be major speed ups.

Reply:
I believe that this code tries to detect if it's a template function or not. Is the string `dostuff` or `dostuff<int>`. We do not know the position of the `<` it's somewhere in the middle of the string. But I guess we could check if the string ends with `>` instead. We have a function `endsWith` that would work well in this case.

In general I am not sure what to think about replacing string literals with char literals in cppcheck code. I have done that before myself but I am not sure about this practice now anymore.

Reply:
> In general I am not sure what to think about replacing string literals with char literals in cppcheck code. I have done that before myself but I am not sure about this practice now anymore.

I think if static code analysis tools give us reasonable suggestions we should probably follow them. After all that you expect people to do the same with the things cppcheck reports. :)

Reply:
users should not blindly follow cppcheck warnings.

we do have some "performance" warnings in cppcheck but I do think this is a questionable category. maybe we should remove these.

Replacing string literals with char literals is a tweak I do not want to have right now... until it is better investigated that there is a clear positive effect.

Reply:
In this particular piece of code I think that using `endsWith` would be better though.

Reply:
Slightly related:
http://0x80.pl/notesen/2016-10-08-slow-std-string-find.html

I saw std::string::find() show up in a few places while profiling. This might explain it.